src/history.rs

```diff
@@ -65,23 +71,36 @@ pub async fn create_controller_revision<T>(
 where
     T: Meta + Hash,
 {
-    let mut hasher = DefaultHasher::new();
-    parent.hash(&mut hasher);
-    let cr = ControllerRevision {
+    let mut cr = ControllerRevision {
         data: Some(data),
         metadata: ObjectMeta {
-            name: Some(controller_revision_name(
-                &Meta::name(parent),
-                &format!("{:x}", hasher.finish()),
-            )),
+            name: None,
             namespace: Meta::namespace(parent),
             owner_references: Some(vec![object_to_owner_reference::<T>(parent.meta().clone())?]),
             ..ObjectMeta::default()
         },
         revision,
     };
 
-    client.create(&cr).await // TODO: Retry logic on conflict?
+    let mut collision_count = 0;
+    loop {
+        let mut hasher = DefaultHasher::new();
+        parent.hash(&mut hasher);
+        collision_count.hash(&mut hasher);
+        let hash = hasher.finish();
+
+        let name = controller_revision_name(&Meta::name(parent), &format!("{:x}", hash));
+        cr.metadata.name = Some(name);
+
+        let result = client.create(&cr).await;
```

Comment:
I have not, no. This is almost 1:1 the behavior from upstream.
This should be "safer" and usually it should only have very few collisions so in the end this might even be cheaper.